### PR TITLE
#24 Migrate to null-safety

### PR DESCRIPTION
**Pull request by jxstxn1**
Addressing Issue #24.
Successfully migrated the project to Null-Safety

Please let me know if I have to do further changes to get the Hacktoberfest Approval

### CODE REVIEW

**Comment by YazeedAlKhalaf** [comment]
will try to check it later today.
thanks for the awesome contribution.

**Comment by jxstxn1** [comment]
No Problem
Thank you for your fast response

File: lib/src/app/models/flutter_installer_api/app_release.mode.dart [hunk 1]
```diff
@@ -5,20 +5,20 @@ import 'package:flutter/foundation.dart';
 import 'package:flutter_installer/src/app/models/flutter_installer_api/download_links.model.dart';
 
 class AppRelease {
-  final String name;
-  final String version;
+  final String? name;
+  final String? version;
   final DownloadLinks downloadLinks;
```
Comment (yahu1031) [inline lib/src/app/models/flutter_installer_api/app_release.mode.dart:8]: can also make this nullable. It's okie if it works fine.

File: lib/src/app/models/flutter_installer_api/latest.model.dart [hunk 2]
```diff
@@ -12,17 +12,17 @@ class Latest {
   final Scripts scripts;
```
Comment (yahu1031) [inline lib/src/app/models/flutter_installer_api/latest.model.dart:10]: here too...

File: pubspec.yaml [hunk 3]
```diff
@@ -4,63 +4,63 @@ version: 0.0.8+8
...
   # Generators
-  build_runner: ^2.0.1
-  auto_route_generator: ^2.0.1
-  injectable_generator: ^1.3.0
+  build_runner: ^2.1.4
+  auto_route_generator: ^2.4.1
+  injectable_generator: ^1.5.2
 
-  msix: ^2.0.0
+  msix: ^2.3.0
```
Comment (yahu1031) [inline pubspec.yaml:59]: [msix](https://pub.dev/packages/msix) config missing.

File: pubspec.yaml [hunk 4]
```diff
@@ -4,63 +4,63 @@ version: 0.0.8+8
...
-  file_selector_macos: ^0.0.4
-  file_selector_linux: ^0.0.2
+  logger: ^1.1.0
+  url_launcher: ^6.0.12
+  http: ^0.13.4
+  file_selector: ^0.8.2+1
+  file_selector_windows: ^0.0.2+1
+  file_selector_macos: ^0.0.4+1
+  file_selector_linux: ^0.0.2+1
   window_size:
```
Comment (yahu1031) [inline pubspec.yaml]: Better to start using [bitsdojo_window](https://pub.dev/packages/bitsdojo_window). We also have frameless windows there.

**Review by yahu1031 — CHANGES_REQUESTED** [review]
@jxstxn1 Just give them a **Try**. And maintain the style of your coding same everywhere.

File: pubspec.yaml (hunk 3 above)
Comment (jxstxn1) [reply pubspec.yaml:59]: Thank you totally missed this. I can only test it on MacOS

File: pubspec.yaml (hunk 4 above)
Comment (jxstxn1) [reply pubspec.yaml]: > [bitsdojo_window](https://pub.dev/packages/bitsdojo_window)

So which packages should I replace with bitsdojo_window?

Comment (yahu1031) [reply pubspec.yaml]: `window_size`

Comment (jxstxn1) [reply pubspec.yaml]: alright

Comment (yahu1031) [reply pubspec.yaml]: not a primary thing but you can just play and see what you can do with it 😄

Comment (jxstxn1) [reply pubspec.yaml]: Yes of course :)

**Comment by jxstxn1** [comment]
@yahu1031 Are you able to test it, because my System is telling me that my Developer Licence doesn't be the Licence which is used in the project.

**Comment by yahu1031** [comment]
lemme try

**Comment by yahu1031** [comment]
@jxstxn1 Better add missing return types for the code. and found this issue

```
Restarted application in 477ms.

════════ Exception caught by widgets library ═══════════════════════════════════
The following assertion was thrown building ValueListenableBuilder<Box<bool>>(dirty, state: _ValueListenableBuilderState<Box<bool>>#44d22):
'package:flutter/src/material/theme_data.dart': Failed assertion: line 350 pos 12: 'colorScheme?.brightness == null || brightness == null || colorScheme!.brightness == brightness': is not true.

The relevant error-causing widget was
ThemeModeBuilder
When the exception was thrown, this was the stack
#2      new ThemeData
#3      App.build.<anonymous closure>
#4      ThemeModeBuilder.build.<anonymous closure>.<anonymous closure>
#5      _ValueListenableBuilderState.build
#6      StatefulElement.build
#7      ComponentElement.performRebuild
#8      StatefulElement.performRebuild
#9      Element.rebuild
#10     ComponentElement._firstBuild
#11     StatefulElement._firstBuild
#12     ComponentElement.mount
...     Normal element mounting (7 frames)
#19     SingleChildWidgetElementMixin.mount
#20     Element.inflateWidget
#21     Element.updateChild
#22     ComponentElement.performRebuild
#23     _InheritedProviderScopeElement.performRebuild
#24     Element.rebuild
#25     ComponentElement._firstBuild
#26     ComponentElement.mount
#27     _InheritedProviderScopeElement.mount
...     Normal element mounting (7 frames)
#34     SingleChildWidgetElementMixin.mount
...     Normal element mounting (21 frames)
#55     Element.inflateWidget
#56     Element.updateChild
#57     RenderObjectToWidgetElement._rebuild
#58     RenderObjectToWidgetElement.mount
#59     RenderObjectToWidgetAdapter.attachToRenderTree.<anonymous closure>
#60     BuildOwner.buildScope
#61     RenderObjectToWidgetAdapter.attachToRenderTree
#62     WidgetsBinding.attachRootWidget
#63     WidgetsBinding.scheduleAttachRootWidget.<anonymous closure>
(elided 13 frames from class _AssertionError, class _RawReceivePortImpl, class _Timer, dart:async, and dart:async-patch)
════════════════════════════════════════════════════════════════════════════════
```

**EDIT(SOLUTION) :**
It is because when you have `colorScheme` property, you can't use `brightness` of a `ThemeData`. use brightness in `colorSheme`.

**Comment by yahu1031** [comment]
@jxstxn1 Sorry to tell you, this cannot be approved AFAIK, It has a big bug. Fix it and your PR will be accepted after the testing 😄 
 If you have discord ping me || Mr.Minnu#1031 ||

**Comment by jxstxn1** [comment]
> @jxstxn1 Sorry to tell you, this cannot be approved AFAIK, It has a big bug. Fix it and your PR will be accepted after the testing 😄

Yes Im fixing the AppTheme bug right now.
If there are any more bugs tell me

**Comment by yahu1031** [comment]
> > @jxstxn1 Sorry to tell you, this cannot be approved AFAIK, It has a big bug. Fix it and your PR will be accepted after the testing 😄
> 
> Yes Im fixing the AppTheme bug right now. If there are any more bugs tell me

@jxstxn1  yes there is. i can't record it(I don't have good app). Ping me in discord or Twitter(find in my profile)

**Comment by jxstxn1** [comment]
> > > @jxstxn1 Sorry to tell you, this cannot be approved AFAIK, It has a big bug. Fix it and your PR will be accepted after the testing 😄
> > 
> > 
> > Yes Im fixing the AppTheme bug right now. If there are any more bugs tell me
> 
> @jxstxn1 yes there is. i can't record it(I don't have good app). Ping me in discord or Twitter(find in my profile)

I pinged you on twitter

**Comment by yahu1031** [comment]
@jxstxn1  The issue is, It builds the UI but it keeps shivering. IDK why?. When I hover, it shows loading hover overall the app and it overlays on all apps. Nothing can be able to do with it. Just run and stop the app. That's it. Better you run the app once.
image

**Comment by YazeedAlKhalaf** [comment]
I am sorry, but I don't have time to review the changes. They are a lot and need a lot of time.

**Comment by yahu1031** [comment]
> I am sorry, but I don't have time to review the changes. They are a lot and need a lot of time.

@YazeedAlKhalaf  LOL, I can handle it but need a fix. This PR is not so stable. It is still buggy. I asked him to check it.

**Comment by YazeedAlKhalaf** [comment]
@jxstxn1
This PR has been sitting here for more than 2 months, I will close it for now but if you can enhance it or split it, it will be great.
Thanks a lot!

**Pull request closed without merging**